custom_components/hubitat/event.py

```diff
@@ -0,0 +1,100 @@
+"""Hubitat event entities."""
+
+from logging import getLogger
+from typing import Unpack
+
+from homeassistant.components.event import (
+    EventDeviceClass,
+    EventEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+
+from .const import (
+    _TRIGGER_ATTRS,
+    _TRIGGER_ATTR_MAP,
+)
+from .device import HubitatEntity, HubitatEntityArgs
+from .hub import get_hub
+from .hubitatmaker import DeviceAttribute, Event
+from .types import EntityAdder
+
+_LOGGER = getLogger(__name__)
+
+
+class HubitatButtonEvent(HubitatEntity, EventEntity):
+    """Representation of an Hubitat button event"""
+
+    _button_id: str
+
+    def __init__(self, button_id: str, **kwargs: Unpack[HubitatEntityArgs]):
+        """Initialize a hubitat button event entity"""
+        HubitatEntity.__init__(self, device_class=EventDeviceClass.BUTTON, **kwargs)
+        EventEntity.__init__(self)
+
+        self._button_id = button_id
+        self._attr_unique_id = f"{super().unique_id}::button_event::{self._button_id}"
+        self._attr_event_types = self.get_event_types()
+
+    @property
+    def name(self) -> str:
```

Comment:
Rather than overriding `name`, set `self._attr_name` in the constructor. That's what the other entity classes _should_ be doing, although not all of them are yet.